### ISSUE TITLE
useless-super-delegation false positive when calling super for __hash__ method

### ISSUE THREAD

**Issue by bboe** [PyCQA/pylint#1647]
### Steps to reproduce

```python
class Parent(object):
    def __hash__(self):
        return 1


class Child(Parent):
    def __eq__(self, _):
        return True

    def __hash__(self):
        return super(Child, self).__hash__()

print(hash(Parent()))
print(hash(Child()))
```

In the previous simplified example, it is necessary to define `__hash__` in such a way on Child in order for instances of `Child` to be hashable using the same method as its Parent class given that its `__eq__` method was overwritten. For reference Python's [data model](https://docs.python.org/3/reference/datamodel.html#object.__hash__) reads:

> A class that overrides `__eq__()` and does not define `__hash__()` will have its `__hash__()` implicitly set to `None`. When the `__hash__()` method of a class is `None`, instances of the class will raise an appropriate `TypeError` when a program attempts to retrieve their hash value, and will also be correctly identified as unhashable when checking `isinstance(obj, collections.Hashable)`.

### Current behavior

Running pylint on this code results in:

    W0235:  10, 4: Useless super delegation in method '__hash__' (useless-super-delegation)

### Expected behavior

No warning should be emitted for such a case.

### pylint --version output

```
pylint 1.7.2, 
astroid 1.5.3
Python 3.6.2 (default, Jul 17 2017, 16:44:45) 
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

**Comment by PCManticore** [comment]
You can rewrite the second ``__hash__`` implementation as ``__hash__ = Parent.__hash__`` and pylint and python will both be happy

**Comment by bboe** [comment]
@PCManticore thanks I wasn't aware. That doesn't necessarily help in the general multiple inheritance case. However, that is not a use case I actually have right now.

**Comment by ticalc-travis** [comment]
I've also found myself in this situation, and in my case it is not convenient to refer to the parent directly by name. I argue that there is nothing wrong with super delegation in the case of __hash__ and that pylint should not complain about it.

**Comment by hippo91** [comment]
@ticalc-travis i'am not sure it is wise to add an exception for ``useless-super-delegation`` in case of the ``__hash__`` method.
The fact that the ``useless-super-delegation`` is a false positive here is a special case because of the redefinition of the ``__eq__`` method. If there was no such redefinition then the message would not be a false positive.
@PCManticore  what do you think about it?
